File: assets/codes/leetcode/src/main/java/leetcode/L_1768.go
package sample

func mergeAlternately(word1 string, word2 string) string {
	word := make([]byte, 0, len(word1)+len(word2))
	i := 0
	for ; i < len(word1) && i < len(word2); i++ {
		word = append(word, word1[i], word2[i])
	}
	if i < len(word1) {
		return string(word) + word1[i:]
	}
	if i < len(word2) {
		return string(word) + word2[i:]
	}
	return string(word)
}

File: assets/codes/leetcode/src/main/java/top75/L_1768.java
package top75;

public class L_1768 {
    public String mergeAlternately(String word1, String word2) {
        StringBuilder sb = new StringBuilder();
        int len1 = word1.length();
        int len2 = word2.length();
        if (len1 == 0 || len2 == 0) {
            return word1 + word2;
        }
        int a = 0;
        int b = 0;
        for (int i = 0; i < len1 + len2; i++) {
            if (a <= b && a < len1 || b >= len2) {
                sb.append(word1.charAt(a));
                a++;
            } else {
                sb.append(word2.charAt(b));
                b++;
            }
        }
        return sb.toString();
    }

    public static void main(String[] args) {

    }
}
